.github/workflows/release.yml

```diff
@@ -0,0 +1,26 @@
+name: Releases
+
+on: 
+  push:
+    tags:
+    - '*'
+
+jobs:
+
+  build:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v1
+    - name: Run tests
+      run: ./gradlew clean test --info
+    - name: Generate build number
+      id: buildnumber
+      uses: einaregilsson/build-number@v3 
+      with:
+        token: ${{secrets.GITHUB_TOKEN}}
+    - name: Build artifacts
+      run: ./gradlew -Penv=production -PbuildNumber=${{ steps.buildnumber.outputs.build_number }} clean assemble
```

Comment:
Why not use the tag name?

Reply:
It's complicated. I didn't want to rework the entire pipeline which depends on the build number. We can't use tags because of the beta channel that substitutes this with `master`, which in turn is not unique. I agree that having this build number fabricated isn't ideal, but babe steps!